aten/src/ATen/cuda/tunable/GemmCommon.h

```diff
@@ -171,4 +179,79 @@ struct GemmStridedBatchedParams : OpParams {
   int64_t batch;
 };
 
+template <typename T>
+struct ScaledGemmParams : OpParams {
+  std::string Signature() const override {
+    return c10::str(transa, transb, "_", m, "_", n, "_", k);
+  }
+
+  ScaledGemmParams* DeepCopy() const {
+    ScaledGemmParams* copy = new ScaledGemmParams;
+    *copy = *this;
+    c10::DeviceIndex device = 0;
+    AT_CUDA_CHECK(c10::cuda::GetDevice(&device));
+    size_t c_size = m * n * sizeof(T);
+    copy->c = c10::cuda::CUDACachingAllocator::raw_alloc(c_size);
+    AT_CUDA_CHECK(c10::cuda::CUDACachingAllocator::memcpyAsync(
+        copy->c, device, c, device, c_size, getCurrentCUDAStream(device), true));
+    return copy;
+  }
+
+  // only call on object returned by DeepCopy
+  void Delete() {
```

Comment:
This seems to ask the user to call this function explicitly before delete the returned pointer of DeepCopy (maybe a potential memory leak if not called). Should there is a DeleteDeepCopy(ScaledGemmParams* ) function to delete the returned pointer by the above DeepCopy() function and delete c inside it?

Reply:
Yes, the contract is if you call DeepCopy(), it is callers responsibility to free it when done by calling Delete().